discovery-provider/src/challenges/challenge_event_bus.py

```diff
@@ -72,25 +58,32 @@ def dispatch(
         user_id: int,
         extra: Dict = {},
     ):
-        """Dispatches an event + block_number + user_id to Redis queue.
+        """Dispatches an event + block_number + user_id to an in memory queue.
 
-        Note: May defer the dispatch if using an in-memory queue (see: use_scoped_dispatch_queue)
+        Does not dispatch to Redis until flush is called or a scoped dispatch queue goes out of scope
         """
-        if self._use_in_memory_queue:
-            logger.debug(
-                "ChallengeEventBus: deferring dispatch - using in-memory queue"
-            )
-            self._in_memory_queue.append(
-                {
-                    "event": event,
-                    "block_number": block_number,
-                    "user_id": user_id,
-                    "extra": extra,
-                }
-            )
-        else:
+        self._in_memory_queue.append(
+            {
+                "event": event,
+                "block_number": block_number,
+                "user_id": user_id,
+                "extra": extra,
+            }
+        )
+
+    def flush(self):
```

Comment:
Don't we need to clear the queue after we send to redis?

Reply:
yes. copy pasting is not my strong suit apparently. 🤦